autopush/websocket.py

```diff
@@ -352,63 +346,40 @@ def process_hello(self, data):
         # This must be a helo, or we kick the client
         cmd = data.get("messageType")
         if cmd != "hello":
-            self.sendClose()
-            return
+            return self.sendClose()
 
         if self.uaid:
-            self.returnError("hello", "duplicate hello", 401)
-            return
+            return self.returnError("hello", "duplicate hello", 401)
 
         uaid = data.get("uaid")
-        valid = False
-        if uaid:
-            try:
-                valid = bool(uuid.UUID(uaid))
-            except ValueError:
-                pass
-        if not uaid or not valid:
-            uaid = str(uuid.uuid4())
+        _, uaid = validate_uaid(uaid)
         self.uaid = uaid
 
-        connect = data.get("connect")
-        if connect and self.ap_settings.bridge:
-            self.transport.pauseProducing()
-            d = self.deferToThread(self.ap_settings.bridge.register, uaid,
-                                   connect)
-            d.addCallback(self._check_router, True)
-            d.addErrback(self.err_hello)
-        else:
-            self._check_router(False)
+        # Default router choice
+        router_type = data.get("router_type", "simplepush")
+        if router_type not in self.ap_settings.routers:
+            return self.returnError("hello", "invalid router", 401)
 
-    def _check_router(self, paused=False, bridge_register=None):
-        if paused:
-            self.transport.resumeProducing()
-        # Bridge registration either succeeds and returns true,
-        # is not specified by the hello, or fails with an exception.
-        router = self.ap_settings.router
-        url = self.ap_settings.router_url
         self.transport.pauseProducing()
-        d = self.deferToThread(router.register_user,
-                               self.uaid, url, self.connected_at)
+        user_item = dict(
+            uaid=self.uaid,
+            node_id=self.ap_settings.router_url,
+            connected_at=self.connected_at,
+            router_type=router_type,
+            router_data={},
```

Comment:
I didn't realize we have clients in the wild that we send UDP wake-ups to, I thought that code was for a partner that implements their own system independent of us. My preference would be for a separate RESTful call for a client to add additional router information (such as a UDP wake-up), mainly because the websocket is an implementation detail of this transport, which will eventually be http2. Having a separate scheme devoted to managing custom router/wake data that is outside other specs feels helpful in that regard, since it means we won't bog down the WebPush spec to add custom wake stuff as well (since we'll have a http endpoint independent of the spec for a client to register that).

To that end, it does present the point that for consistency, we should probably keep the websocket simplepush spec for messages like it is now, and if a client wants webpush style independent message delivery, it can make the HTTP call first to get a UAID and ask for it, then do the websocket connection. And no ``router_type`` option will be available during the websocket connection.